src/_data/glossary.yml

```
@@ -730,6 +730,55 @@
   alternate:
     - "child class"
 
+- term: "Shadowing"
+  short_description: |-
+    When a local declaration hides another with the same name.
+  long_description: |-
+    **Shadowing** occurs when a local declaration (such as a variable or parameter)
+    uses the same name as an existing declaration in an outer scope,
+    making the outer one inaccessible within the inner scope.
+
+    Shadowing is valid in Dart, but it can lead to confusing or unintended behavior,
+    so it’s generally discouraged unless used deliberately and clearly.
+
+    ### Example
+
+    ```dart
+    String message = 'Global';
```

Unless a sample is specifically related to typing a variable, we prefer to use type inference for local variables as well as for top-level variables with relatively "[obvious](https://dart.dev/tools/linter-rules/omit_obvious_property_types)".

```suggestion
    final message = 'Global';
```